File: external/HDF-EOS/hdfeos/samples/inquireswath.f

c
c	In this example we will retrieve (1) information about the
c	dimensions, (2) the dimension mappings (geolocation relations), 
c	and (3) the swath fields.
c 

	program inquireswath

	integer         i, swmapinfo, swfldinfo, status
	integer         swdetach, swclose
	integer*4       swinqdims, swinqmaps, swinqdflds, swinqgflds
	integer*4       swdiminfo, offset, incr, rk, nt
	integer*4       swimapinfo, swinqimaps
	integer*4       swfid, swid, ndims, nmaps, nflds
	integer*4       swopen, swattach
	integer*4       dims(32), off(32), inc(32), rank(32), ntype(32)
	integer*4       sizes(8), indx(32), dimsize
	character*72    dimname, dimmap, fieldlist, dimlist
	integer         DFACC_READ
	parameter        (DFACC_READ=1)

c
c	Open the Swath File for read only access
c

	swfid = swopen("SwathFile.hdf", DFACC_READ)

    
	if (swfid .NE. -1) then

c	   Attach the swath

	   swid = swattach(swfid, "Swath1")
	
	   if (swid .NE. -1) then


c	    Inquire Dimensions

	      ndims = swinqdims(swid, dimname, dims)

	      write(*,*) 'Dimension list: ', dimname
	      do i = 1,ndims
		 write(*,*) 'dim size: ', dims(i)
	      enddo
	      write(*,*)	      


c           Inquire Dimension Mappings

	      nmaps = swinqmaps(swid, dimmap, off, inc)

	      write(*,*) 'Dimension map: ', dimmap
	      do i = 1,nmaps
		 write(*,*) 'offset increment: ', off(i), inc(i)
	      enddo
	      write(*,*)

	      
	      
c	    Inquire Indexed Dimension Mappings

	      nmaps = swinqimaps(swid, dimmap, sizes)

	      write(*,*) 'Index Dimension map: ', dimmap
	      do i=1,nmaps
		 write(*,*) 'sizes: ', sizes(i)
	      enddo
	      write(*,*)	      

	     
c         Inquire Geolocation Fields
	    
	      nflds = swinqgflds(swid, fieldlist, rank, ntype)
	      write(*,*) 'Geolocation fieldlist: ', fieldlist
	      do i=1,nflds
		 write(*,*) 'field rank & datatype: ', rank(i), ntype(i)
	      enddo
	      write(*,*) 
	    
	      
c         Inquire Data Fields	      

	      nflds = swinqdflds(swid, fieldlist, rank, ntype)
	      write(*,*) 'Data Fieldlist: ', fieldlist
	      do i=1,nflds
		 write(*,*) 'field rank & datatype: ', rank(i), ntype(i)
	      enddo
	      write(*,*) 
	    



c	  Get info on "GeoTrack" dim

	      dimsize = swdiminfo(swid, "GeoTrack")
	      write(*,*) 'Size of GeoTrack: ', dimsize
	      write(*,*)
	      

c	  Get info on "GeoTrack/Res2tr" mapping

	      status = swmapinfo(swid, "GeoTrack", "Res2tr", offset,
     1                           incr)	      
	      write(*,*) 'Mapping Offset: ', offset
	      write(*,*) 'Mapping Increment: ', incr
	      write(*,*)


c	  Get info on "IndxTrack/Res2tr" indexed mapping

	      n = swimapinfo(swid, "IndxTrack", "Res2tr", indx)
	      do i=1,n
		 write(*,*) 'Index Mapping Entry ', i, indx(i)
	      enddo
	      write(*,*)
	      

c	  Get info on "Longitude" Field

	      status = swfldinfo(swid, "Longitude", rk, dims, nt, dimlist)
	      write(*,*) 'Longitude Rank: ', rk
	      write(*,*) 'Longitude NumberType: ', nt
	      write(*,*) 'Longitude Dimlist: ', dimlist
	      do i=1,rk
		 write(*,*) 'Dimension ',i,dims(i)
	      enddo
	     
	  endif
	endif
	
	status = swdetach(swid)
	status = swclose(swfid)
	stop
	end

File: external/HDF-EOS/hdfeos/samples/InquireSwath.c
#include "hdf.h"
#include "HdfEosDef.h"

/*
 * In this example we will retrieve (1) information about the dimensions, (2)
 * the dimension mappings (geolocation relations), and (3) the swath fields.
 */


main()
{

    intn            status, i;
    int32           swfid, SWid, ndims, nmaps, rk, nt, dim[8], nflds;
    int32          *dims, *off, *inc, *indx, *rank, *ntype;
    int32           n, strbufsize, dimsize, offset, incr, *sizes;

    char           *dimname, *dimmap, *fieldlist, dimlist[80];


    /*
     * Open the Swath File for read only access
     */

    swfid = SWopen("SwathFile.hdf", DFACC_READ);


    if (swfid != -1)
    {

	/* Attach the swath */

	SWid = SWattach(swfid, "Swath1");

	if (SWid != -1)
	{
	    /* Inquire Dimensions */
	    ndims = SWnentries(SWid, HDFE_NENTDIM, &strbufsize);
	    dims = (int32 *) calloc(ndims, 4);
	    dimname = (char *) calloc(strbufsize + 1, 1);
	    
	    ndims = SWinqdims(SWid, dimname, dims);

	    printf("Dimension list: %s\n", dimname);
	    for (i = 0; i < ndims; i++)
		printf("dim size: %d\n", dims[i]);
	    
	    free(dims);
	    free(dimname);


	    /* Inquire Dimension Mappings */
	    nmaps = SWnentries(SWid, HDFE_NENTMAP, &strbufsize);
	    
	    off = (int32 *) calloc(nmaps, 4);
	    inc = (int32 *) calloc(nmaps, 4);
	    dimmap = (char *) calloc(strbufsize + 1, 1);
	    
	    nmaps = SWinqmaps(SWid, dimmap, off, inc);
	    printf("Dimension map: %s\n", dimmap);
	    for (i = 0; i < nmaps; i++)
		printf("offset increment: %d %d\n",
		       off[i], inc[i]);
	    free(off);
	    free(inc);
	    free(dimmap);


	    /* Inquire Indexed Dimension Mappings */
	    nmaps = SWnentries(SWid, HDFE_NENTIMAP, &strbufsize);
	    sizes = (int32 *) calloc(nmaps, 4);
	    dimmap = (char *) calloc(strbufsize + 1, 1);
	    nmaps = SWinqidxmaps(SWid, dimmap, sizes);

	    printf("Index Dimension map: %s\n", dimmap);
	    for (i = 0; i < nmaps; i++)
		printf("sizes: %d\n", sizes[i]);

	    free(sizes);
	    free(dimmap);


	    /* Inquire Geolocation Fields */
	    nflds = SWnentries(SWid, HDFE_NENTGFLD, &strbufsize);
	    rank = (int32 *) calloc(nflds, 4);
	    ntype = (int32 *) calloc(nflds, 4);
	    fieldlist = (char *) calloc(strbufsize + 1, 1);
	    nflds = SWinqgeofields(SWid, fieldlist, rank, ntype);

	    printf("geo fields: %s\n", fieldlist);
	    for (i = 0; i < nflds; i++)
		printf("rank type: %d %d\n", rank[i], ntype[i]);

	    free(rank);
	    free(ntype);
	    free(fieldlist);



	    /* Inquire Data Fields */
	    nflds = SWnentries(SWid, HDFE_NENTDFLD, &strbufsize);
	    rank = (int32 *) calloc(nflds, 4);
	    ntype = (int32 *) calloc(nflds, 4);
	    fieldlist = (char *) calloc(strbufsize + 1, 1);
	    nflds = SWinqdatafields(SWid, fieldlist, rank, ntype);

	    printf("data fields: %s\n", fieldlist);
	    for (i = 0; i < nflds; i++)
		printf("rank type: %d %d\n", rank[i], ntype[i]);

	    free(rank);
	    free(ntype);
	    free(fieldlist);


	    /* Get info on "GeoTrack" dim */
	    dimsize = SWdiminfo(SWid, "GeoTrack");
	    printf("Size of GeoTrack: %d\n", dimsize);


	    /* Get info on "GeoTrack/Res2tr" mapping */
	    status = SWmapinfo(SWid, "GeoTrack", "Res2tr", &offset, &incr);
	    printf("Mapping Offset: %d\n", offset);
	    printf("Mapping Increment: %d\n", incr);


	    /* Get info on "IndxTrack/Res2tr" indexed mapping */
	    dimsize = SWdiminfo(SWid, "IndxTrack");
	    indx = (int32 *) calloc(dimsize, 4);
	    n = SWidxmapinfo(SWid, "IndxTrack", "Res2tr", indx);
	    for (i = 0; i < n; i++)
		printf("Index Mapping Entry %d: %d\n", i+1, indx[i]);
	    free(indx);


	    /* Get info on "Longitude" Field */
	    status = SWfieldinfo(SWid, "Longitude", &rk, dim, &nt, dimlist);
	    printf("Longitude Rank: %d\n", rk);
	    printf("Longitude NumberType: %d\n", nt);
	    printf("Longitude Dimension List: %s\n", dimlist);	    
	    for (i=0; i<rk; i++)
		printf("Dimension %d: %d\n",i+1,dim[i]);

	}
    }
    status = SWdetach(SWid);
    status = SWclose(swfid);

    return;
}
